### PR TITLE
Downgrade warning about uninitialized-/etc in tmpfiles

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/tmpfiles/tmpfiles.c [hunk 1]
```diff
@@ -225,17 +225,11 @@ static const char *const creation_mode_verb_table[_CREATION_MODE_MAX] = {
...
-        r = specifier_machine_id(specifier, data, root, userdata, ret);
-        if (IN_SET(r, -ENOENT, -ENOMEDIUM))
-                return -ENXIO;
-
-        return r;
+static inline bool ERRNO_IS_NOINFO(int r) {
+        return IN_SET(abs(r),
+                      EUNATCH,
+                      ENOMEDIUM,
+                      ENXIO);
```
Comment (poettering) [inline src/tmpfiles/tmpfiles.c]: hmm, this sounds very specifiy to xdg_user_config_dir(), no?

this really needs some comments, in particular where we'll see each error, and why it makes sense to handle this all as the same case

File: src/tmpfiles/tmpfiles.c [hunk 2]
```diff
@@ -3435,9 +3429,10 @@ static int parse_line(
                         return log_syntax(NULL, LOG_ERR, fname, line, SYNTHETIC_ERRNO(EINVAL), "Credential name not valid: %s", i.argument);
 
                 r = read_credential(i.argument, &i.binary_argument, &i.binary_argument_size);
-                if (IN_SET(r, -ENXIO, -ENOENT)) {
+                if (ERRNO_IS_NOINFO(r)) {
```
Comment (poettering) [inline src/tmpfiles/tmpfiles.c]: this drop the check abotu ENOENT?

File: src/tmpfiles/tmpfiles.c [hunk 3]
```diff
@@ -290,14 +290,18 @@ static int log_unresolvable_specifier(const char *filename, unsigned line) {
          * not considered as an error so log at LOG_NOTICE only for the first time
          * and then downgrade this to LOG_DEBUG for the rest. */
 
+        int log_level = notified || arg_root || sd_booted() == 0 ?
+                LOG_DEBUG : LOG_NOTICE;
```
Comment (poettering) [inline src/tmpfiles/tmpfiles.c:300]: please add discussion of the sd_booted() check into comments here...

why not use a running_in_chroot() check here?

**Review by bluca — APPROVED** [review]

File: src/tmpfiles/tmpfiles.c (hunk 2 above)
Comment (keszybz) [reply src/tmpfiles/tmpfiles.c]: I dropped that hunk. As you said, it dropped the check for ENOENT, and I don't think it should be handled the same as the others.

**Comment by keszybz** [comment]
Updated:
- more comments
- I changed `sd_booted()` to `running_in_chroot()`. I think both should give pretty much the same result.

**Review by yuwata — APPROVED** [review]

**Pull request merged**